src/modules/Popup/Popup.js

```diff
@@ -0,0 +1,302 @@
+      if (style.right + pageXOffset < 0) {
+        return 'left center'
+      }
+    } else if (style.left - pageXOffset < 0) {
+      return 'right center'
+    }
+
+    if (_.isNumber(style.bottom)) {
+      if (style.bottom + this.popupCoords.height > clientHeight - pageYOffset) {
+        return 'bottom center'
+      }
+    } else if (style.top + this.popupCoords.height > pageYOffset + clientHeight) {
+      return 'top center'
+    }
+  }
+
+  getPopupStyle() {
+    if (!this.coords || !this.popupCoords) return
+    let positioning = this.props.positioning
+    let style = this.computePopupStyle(positioning)
+
+    // Lets detect if the popup is out of the viewport and adjust
+    // the position accordingly
+    const correctedPosition = this.correctPositionForViewPort(style)
+    if (correctedPosition) {
+      style = this.computePopupStyle(correctedPosition)
+      // Set the position so the carret will appear at the right place
+      positioning = correctedPosition
+    }
+
+    // Append 'px' to every numerical values in the style
+    style = _.mapValues(style, value => _.isNumber(value) ? value + 'px' : value)
+
+    this.setState({ style, positioning })
+  }
+
+  getPortalProps() {
+    const portalProps = { onOpen: this.onOpen }
+    const { on, hoverable } = this.props
+
+    switch (on) {
+      case 'click':
+        portalProps.openOnTriggerClick = true
+        portalProps.closeOnTriggerClick = true
+        portalProps.closeOnDocumentClick = true
+        break
+
+      case 'focus':
+        portalProps.openOnTriggerFocus = true
+        portalProps.closeOnTriggerBlur = true
+        break
+
+      default:  // default to hover
+        portalProps.openOnTriggerMouseOver = true
+        portalProps.closeOnTriggerMouseLeave = true
+        // Taken from SUI: https://git.io/vPmCm
+        portalProps.mouseLeaveDelay = 70
+        portalProps.mouseOverDelay = 50
+        break
+    }
+
+    if (hoverable) {
+      portalProps.closeOnPortalMouseLeave = true
+      portalProps.mouseLeaveDelay = 300
+    }
+
+    return portalProps
+  }
+
+  hideOnScroll = (event) => {
+    this.setState({ closed: true })
+    window.removeEventListener('scroll', this.hideOnScroll)
+    setTimeout(() => this.setState({ closed: false }), 50)
+  }
+
+  onOpen = (event) => {
+    this.coords = event.currentTarget.getBoundingClientRect()
+    if (this.props.hideOnScroll) {
+      window.addEventListener('scroll', this.hideOnScroll)
+    }
+  }
+
+  popupMounted = (ref) => {
+    this.popupCoords = ref ? ref.getBoundingClientRect() : null
+    this.getPopupStyle()
+  }
+
+  render() {
+    const {
+      basic,
+      children,
+      className,
+      content,
+      flowing,
+      header,
+      inverted,
+      size,
+      trigger,
+      wide,
+    } = this.props
+
+    const { positioning, closed } = this.state
+
+    const style = _.assign({}, this.state.style, this.props.style)
+
+    const classes = cx(
```

Comment:
Update usage of `cx` to follow the standard pattern of this lib (see https://github.com/TechnologyAdvice/stardust/blob/master/src/collections/Grid/Grid.js). There are helpers to set the appropriate class based on the prop.

For example, `wide` here can use the useKeyOrValueAndKey (see similar prop usage: https://github.com/TechnologyAdvice/stardust/blob/master/src/collections/Grid/Grid.js#L34-L35)